src/csm_testing/tests/iuf_pre_checks/__main__.py

```diff
@@ -0,0 +1,554 @@
+            print("ERROR: Some pods are not running")
+            print(cfs_ara_pods)
+            sys.exit(1)
+        else:
+            print(f"ERROR: Return code {result.returncode}")
+            print(cfs_ara_pods)
+            sys.exit(1)
+    except subprocess.CalledProcessError:
+        print("INFO: All cfs-ara-postgres* pods are running")
+    except Exception:
+        sys.exit(1)
+
+
+def check_sat():
+    """
+    Function to check sat status and print the info.
+    Args:
+        None
+    Returns:
+        None
+    """
+    sat, returncode = run_command("sat --version")
+    if returncode == 0:
+        print(f"INFO: sat is installed. {sat}")
+    else:
+        print(
+            "ERROR: Unable to run sat.Please verify if sat is installed and working"
+        )
+        sys.exit(1)
+
+
+def check_docs_and_libcsm():
+    """
+    Function to check docs-csm and lib-csm rpms and print the info.
+    Args:
+        None
+    Returns:
+        None
+    """
+    docs_csm, returncode = run_command("rpm -qa | grep docs-csm")
+    if returncode == 0:
+        print(f"INFO: docs-csm rpm is installed. Version: {docs_csm}")
+    else:
+        print(
+            "ERROR: Unable to find docs. Please verify if docs is installed.")
+        sys.exit(1)
+    lib_csm, returncode = run_command("rpm -qa | grep docs-csm")
+    if returncode == 0:
+        print(f"INFO: lib-csm is installed. Version: {lib_csm}")
+    else:
+        print(
+            "ERROR: Unable to find libcsm. Please verify if libcsm is installed."
+        )
+        sys.exit(1)
+
+
+def check_cpc():
+    """
+    Function to check cray-product-catalog configmap and print the info.
+    Args:
+        None
+    Returns:
+        None
+    """
+    _, returncode = run_command(
+        "kubectl get cm -n services | grep cray-product-catalog")
+    if returncode == 0:
+        print(
+            "INFO: Cray-Product-Catalog configmap exists in services namespace"
+        )
+    else:
+        print("ERROR: Unable to find Cray-Product-Catalog configmap.")
+        sys.exit(1)
+
+
+def check_available_space():
+    """
+    Function to check available disk space and print the info.
+    Args:
+        None
+    Returns:
+        None
+    """
+    _, _, free = shutil.disk_usage("/etc/cray/upgrade/csm/")
+    free_gb = free // (2**30)
+    if free_gb < 5:
+        print("ERROR: Insufficient space in /etc/cray/upgrade/csm/.")
+        sys.exit(1)
+    else:
+        print(f"INFO: Available space: {free_gb}G")
+
+
+def check_url_status(cluster_name: str):
+    """
+    Function to check argo,vcs,nexus reachability and print the info.
+    Args:
+        cluster_name(str): Cluster name
+    Returns:
+        None
+    """
+    urls = [
+        f"https://argo.cmn.{cluster_name}.hpc.amslabs.hpecorp.net/",
+        f"https://vcs.cmn.{cluster_name}.hpc.amslabs.hpecorp.net/",
+        f"https://nexus.cmn.{cluster_name}.hpc.amslabs.hpecorp.net/",
+    ]
```

Comment:
hpc.amslabs.hpecorp.net is an internal domain, no customer system will have these values.

One way would be to read the SYSTEM_DOMAIN environment variable if it's available here. Another would be to interrogate Kubernetes for the list of virtual services (`kubectl get virtualservice -A`)